Source/WebKit/UIProcess/WebAuthentication/fido/U2fAuthenticator.cpp

```diff
@@ -157,7 +157,7 @@ void U2fAuthenticator::continueRegisterCommandAfterResponseReceived(ApduResponse
     case ApduResponse::Status::SW_NO_ERROR: {
         auto& options = std::get<PublicKeyCredentialCreationOptions>(requestData().options);
         auto appId = processGoogleLegacyAppIdSupportExtension(options.extensions);
-        auto response = readU2fRegisterResponse(!appId ? options.rp.id : appId, apduResponse.data(), AuthenticatorAttachment::CrossPlatform, { driver().transport() }, options.attestation);
+        auto response = readU2fRegisterResponse(!appId ? *options.rp.id : appId, apduResponse.data(), AuthenticatorAttachment::CrossPlatform, { driver().transport() }, options.attestation);
```

Comment:
Since `options.rp.id` may now be `std::nullopt`, do we need to do any validation here? Maybe at least ASSERT so we catch cases where a null opt got through?

Reply:
It should get replaced in AuthenticatorCoordinator when `std::nullopt`. I will add ASSERTs.